### PR TITLE
Windows upgrade and testing [WIP]

### PR DESCRIPTION
**Pull request by fatz**
fixes #101 

- added update tests
- wrong variable used
- facts are missing

Including facts does not work yet

### CODE REVIEW

**Comment by janisz** [comment]
> flaky ansible tests where related to spot instance capacity. As ansible is not able to give a spot block time and the capacity seems to be pretty low. AWS kills some of our test instances from time to time in the middle of a test.

> This gives us pretty weird errors as ansible tasks time out or a first taks running fine and a second one throws SSH timeout.

> The current workaround is deactivating spot usage. But we need to investigate how to let ansible use spot block so we can request spot instances with a minimum running time
https://mesosphere.slack.com/archives/CHPEMR8VA/p1585156668020900?thread_ts=1585156668.020900&cid=CHPEMR8VA

**Review by janisz — APPROVED** [review]

File: roles/dcos_agent_windows/tasks/main.yml [hunk 1]
```diff
@@ -8,23 +13,27 @@
     path: '{{ base_windows_dir }}\etc\dcos-version.json'
   register: previous_installation
 
+- name: Windows | Create dcos_installation.fact.ps1
+  win_copy:
+    content: get-content {{ base_windows_dir }}\etc\ansible\facts.d\dcos_installation.fact -Raw | ConvertFrom-Json
```
Comment (sergiimatusEPAM) [inline roles/dcos_agent_windows/tasks/main.yml]: the content of dcos_installaton.fact file will be available after task "Copy dcos-version.json to fact directory" . Therefore I don't recommend line 16 .. 22 to be added with anything

File: roles/dcos_agent_windows/tasks/main.yml [hunk 2]
```diff
@@ -8,23 +13,13 @@
...
   win_copy:
-    content: get-content {{ base_windows_dir }}\etc\ansible\facts.d\dcos_installation.fact -Raw | ConvertFrom-Json
+    content: get-content {{ base_windows_dir }}\\etc\\dcos-version.json -Raw | ConvertFrom-Json
     dest: "{{ base_windows_dir }}\\etc\\ansible\\facts.d\\dcos_installation.fact.ps1"
   when: previous_installation.stat.exists and previous_installation.stat.isreg
-
-- name: Windows | Re-run setup to use custom facts
-  setup:
-    fact_path: "{{ base_windows_dir }}\\etc\\ansible\\facts.d\\dcos_installation.fact.ps1"
-  when: previous_installation.stat.exists and previous_installation.stat.isreg
```
Comment (sergiimatusEPAM) [inline roles/dcos_agent_windows/tasks/main.yml:27]: @fatz , 
Task "Re-run setup to use custom facts" with `setup` module cannot be deleted as it requires `fact_path` argument. Realization of `setup` module is different for Linux vs Windows. You need either leave the task here, or move the block into handlers/main.yml

Comment (fatz) [reply roles/dcos_agent_windows/tasks/main.yml:27]: yep indeed... that was an mistake...

**Comment by fatz** [comment]
as this deactivates spot instance usage I've created #107 tracking this as a bug

**Pull request merged**